build.proj

```diff
@@ -65,7 +65,23 @@
 
   <Import Project="dir.targets" />
   <Import Project="dir.traversal.targets" />
-  <Import Project="$(ToolsDir)clean.targets" />
+  <Import Project="$(ToolsDir)clean.targets" />  
+ 
+  <!-- 
+    This is to initialize the ProjectJsonFiles ItemGroup with the template files during the update dependencies target
+    only to prevent ValidateExactRestore from failing on just the regular project.json files.
+  -->
+  <Target Name="IncludeOnlyTemplateFilesForUpdate" BeforeTargets="UpdateDependencies">
+    <ItemGroup>
+      <ProjectJsonFiles Remove="*" />
```

Comment:
Although I'm still not sure why we need to do this in a target? Why can't we control what we are putting into this and not need to all Remove?